src/Controls/tests/Core.UnitTests/StyleTests.cs

```diff
@@ -1011,5 +1006,186 @@ public void ClearValueShouldntUnapplyStyles()
+
+			stackLayout.Children.Add(label);
+			Assert.Equal("implicit", label.Text);
+
+			label.Style = style;
+			Assert.Equal("style", label.Text);
+
+			label.Text = "value";
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.StyleProperty);
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.TextProperty);
+			Assert.Equal("implicit", label.Text);
+		}
+
+		[Fact]
+		public void UnapplyingStyleDefaultToImplicit2()
+		{
+			var stackLayout = new StackLayout
+			{
+				Resources = new ResourceDictionary {
+					new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "implicit" } } },
+				}
+			};
+			var label = new Label();
+			var style = new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "style" }, } };
+
+			Assert.Equal(label.Text, Label.TextProperty.DefaultValue);
+
+			label.Style = style;
+			Assert.Equal("style", label.Text);
+
+			stackLayout.Children.Add(label);
+			Assert.Equal("style", label.Text);
+
+			label.Text = "value";
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.StyleProperty);
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.TextProperty);
+			Assert.Equal("implicit", label.Text);
+		}
+
+		[Fact]
+		public void UnapplyingStyleDefaultToImplicit3()
+		{
+			var stackLayout = new StackLayout
+			{
+				Resources = new ResourceDictionary {
+					new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "implicit" } } },
+				}
+			};
+			var label = new Label();
+			var style = new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "style" }, } };
+
+			Assert.Equal(label.Text, Label.TextProperty.DefaultValue);
+
+			label.Text = "value";
+			Assert.Equal("value", label.Text);
+
+			label.Style = style;
+			Assert.Equal("value", label.Text);
+
+			stackLayout.Children.Add(label);
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.StyleProperty);
+			Assert.Equal("value", label.Text);
+
+			label.ClearValue(Label.TextProperty);
+			Assert.Equal("implicit", label.Text);
+		}
+
+		[Fact]
+		public void UnapplyingStyleDefaultToImplicit4()
+		{
+			var stackLayout = new StackLayout
+			{
+				Resources = new ResourceDictionary {
+					new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "implicit" } } },
+				}
+			};
+			var label = new Label();
+			var style = new Style(typeof(Label)) { Setters = { new Setter { Property = Label.TextProperty, Value = "style" }, } };
+
+			Assert.Equal(label.Text, Label.TextProperty.DefaultValue);
+
+			stackLayout.Children.Add(label);
+			Assert.Equal("implicit", label.Text);
+
+			label.Style = style;
+			Assert.Equal("style", label.Text);
+
+			label.ClearValue(Label.StyleProperty);
+			Assert.Equal("implicit", label.Text);
+		}
+
+		[Fact]
+		public void UnapplyingStyleDefaultToImplicit5()
```

Comment:
>UnapplyingStyleDefaultToImplicit5 [](http://example.com/codeflow?start=14&length=33)

As discussed.  We should probably add test case for BasedOn.  

Apply a Style 'foo' back on 'foobase',  'foobase' , remove 'foobase' as a basedon style from 'foo'

Reply:
done